enterprise/server/backends/migration_cache/migration_cache.go

```diff
@@ -232,9 +233,15 @@ func (mc *MigrationCache) FindMissing(ctx context.Context, resources []*resource
 	// Some implementations of FindMissing sort the resources slice, which can cause a race condition if both
 	// the src and dest cache try to sort at the same time. Copy the slice to prevent this
 	var resourcesCopy []*resource.ResourceName
+	var hashToResource map[string]*resource.ResourceName
 	if doubleRead {
 		resourcesCopy = make([]*resource.ResourceName, len(resources))
 		copy(resourcesCopy, resources)
+
+		hashToResource = make(map[string]*resource.ResourceName)
+		for _, r := range resources {
+			hashToResource[r.GetDigest().GetHash()] = r
+		}
```

Comment:
This change LGTM but seems unrelated -- just leaving this comment as a sanity check in case you meant to include it in another PR instead.

Reply:
Thanks for the call out. I'll add the following note to the commit description too to reduce confusion

I changed this because I realized I was hardcoding "Compressor: repb.Compressor_IDENTITY" when queuing the data to be copied below. FindMissing is always called with resources with the identity compressor anyway, so this shouldn't matter. But it's technically incorrect and could cause a subtle bug later